resolvers.py

```diff
@@ -0,0 +1,1979 @@
+            qs_filtered = models.OverallBenchByCapability.objects.filter(
+                capability_name__in=filters)
+            qs_sum = qs_filtered.filter(band_name='Total').aggregate(bench=Sum('bench'), lop=Sum('lop'), resigned=Sum('resigned'), maternity=Sum('maternity'), allocation_pending=Sum('allocation_pending'), nominated=Sum(
+                'nominated'), non_deployable_total=Sum('non_deployable_total'), deployable_total=Sum('deployable_total'), sabbatical=Sum('sabbatical'), grand_total=Sum('grand_total'), grand_total_percentage=Sum('grand_total_percentage'))
+            qs_dummy = {"id": "Total",
+                        "capability_name": 'Total', "band_name": 'Total'}
+            qs_dummy.update(qs_sum)
+            qs = list(chain(qs_filtered, [qs_dummy]))
+        return qs
+
+
+class PeopleOnBenchHelper():
+
+    def __init__(self):
+        today = date.today()
+        # filter added
+        self.qs = models.BaseBenchDataSource.objects.filter((Q(on_bench__iexact='true') | Q(
+            on_bench__iexact='n/a')) & Q(department__iexact='delivery') & Q(date_of_joining__lte=today) & Q(band__in=['B5H', 'B5L', 'B6H', 'B6L', 'B7', 'B8']))
+
+    def add_band_filter(self, band_name):
+        if 'B8 (Intern)' in band_name:
+            if (len(band_name) == 1):
+                return self.qs.filter((Q(band='B8') & Q(designation='Intern')))
+            else:
+                qs_intern = self.qs.filter(
+                    (Q(band='B8') & Q(designation='Intern')))
+                qs_rest = self.qs.filter(
+                    Q(band__in=band_name) & ~Q(designation='Intern'))
+                return qs_intern | qs_rest
+        elif 'Total' in band_name:
+            return self.qs
+        else:
+            return self.qs.filter(Q(band__in=band_name) & ~Q(designation='Intern'))
+
+    def add_capability_filter(self, capability_name, band_name):
+        if 'Total' in capability_name:
+            return self.qs
+        elif 'Interns' in capability_name and (len(capability_name) == 1):
+            return self.add_intern_filter(band_name)
+        elif 'Other' in capability_name and (len(capability_name) == 1):
+            return self.add_other_capability_filter(band_name)
+        elif (len(capability_name) == 1):
+            return self.add_capability_name_filter(band_name, capability_name)
+        else:
+            return self.add_multilpe_filters(capability_name)
+
+    def add_multilpe_filters(self, capability_name):
+        qs_rest = self.qs.filter(
+            Q(capabilities__in=capability_name) & ~Q(designation='Intern'))
+        if 'Interns' in capability_name:
+            capability_name.remove('Interns')
+            qs = self.add_intern_filter('Total')
+            qs_rest = qs_rest | qs
+        if 'Other' in capability_name:
+            capability_name.remove('Other')
+            qs = self.add_other_capability_filter('Total')
+            qs_rest = qs_rest | qs
+        if (len(capability_name) > 0):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)